sklearn/metrics/_plot/roc_curve.py

```diff
@@ -333,20 +382,437 @@ def from_predictions(
+        --------
+        >>> import matplotlib.pyplot as plt
+        >>> from sklearn.datasets import make_classification
+        >>> from sklearn.metrics import RocCurveDisplay
+        >>> from sklearn.model_selection import cross_validate
+        >>> from sklearn.svm import SVC
+        >>> X, y = make_classification(random_state=0)
+        >>> clf = SVC(random_state=0)
+        >>> cv_results = cross_validate(
+        ...     clf, X, y, cv=3, return_estimator=True, return_indices=True)
+        >>> RocCurveDisplay.from_cv_results(cv_results, X, y, kind="both")
+        <...>
+        >>> plt.show()
+        """
+        check_matplotlib_support(f"{cls.__name__}.from_cv_results")
+
+        required_keys = {"estimator", "indices"}
+        if not all(key in cv_results for key in required_keys):
+            raise ValueError(
+                "cv_results does not contain one of the following required keys: "
+                f"{required_keys}. Set explicitly the parameters return_estimator=True "
+                "and return_indices=True to the function cross_validate."
+            )
+
+        train_size, test_size = (
+            len(cv_results["indices"]["train"][0]),
+            len(cv_results["indices"]["test"][0]),
         )
 
-        return viz.plot(ax=ax, name=name, **kwargs)
+        if _num_samples(X) != train_size + test_size:
+            raise ValueError(
+                "X does not contain the correct number of samples. "
+                f"Expected {train_size + test_size}, got {_num_samples(X)}."
+            )
+
+        # TODO: it should be remove once #23073 is merged since we delegate this check
+        # to `_get_response_values_binary`.
+        if not all(is_classifier(estimator) for estimator in cv_results["estimator"]):
+            raise ValueError(
+                "The estimators in cv_results['estimator'] must be fitted classifiers."
+            )
+
+        if fold_name is None:
+            # create an iterable of the same length as the number of ROC curves
+            fold_name_ = [None] * len(cv_results["estimator"])
+        elif fold_name is not None and len(fold_name) != len(cv_results["estimator"]):
+            raise ValueError(
+                "When `fold_name` is provided, it must have the same length as "
+                f"the number of ROC curves to be plotted. Got {len(fold_name)} names "
+                f"instead of {len(cv_results['estimator'])}."
+            )
+        else:
+            fold_name_ = fold_name
+
+        displays = []
+        for fold_id, (estimator, test_indices, name) in enumerate(
+            zip(cv_results["estimator"], cv_results["indices"]["test"], fold_name_)
+        ):
+            y_true = _safe_indexing(y, test_indices)
+            y_pred = _get_response(
+                _safe_indexing(X, test_indices),
+                estimator,
+                response_method=response_method,
+                pos_label=pos_label,
+            )[0]
+
```

Comment:
This makes me wonder if it's the right approach. The user has to provide exactly the same X as the one used to compute the cross val and we have no way to ensure that. For instance if the dataset is shuffled in between, the results will be overconfident due to leakage.

Would it make sense to not accept the cv_results, but compute the cross val as part of this method ?

Reply:
You mean to add a `return_predictions` to `cross_validation` and directly get the prediction instead of the indices?

Reply:
No, I meant run the cross-validation inside the display, but giving it more thoughts it's probably not a good idea: we don't want the display to fit the estimators.

I remember that we talked about returning the prediction from cross_validate, but it was not necessarily easy. I can't remember why, can you remind me ?

Reply:
> I remember that we talked about returning the prediction from cross_validate, but it was not necessarily easy. I can't remember why, can you remind me ?

It is just that this is not implemented but we could make `cross_validate` return the predictions. However, we should also know the method that provides the predictions.